attest/verifier/build.rs

```diff
@@ -290,4 +81,170 @@ fn main() {
+        u8::try_from(start_time.hour()).expect("Hour not a u8"),
+        u8::try_from(start_time.minute()).expect("Minute not a u8"),
+        u8::try_from(start_time.second()).expect("Second not a u8"),
+    )
+    .expect("Could not create a not_before time");
+    let signer_not_after = Time::new(
+        u16::try_from(end_time.year()).expect("Year not a u16"),
+        u8::try_from(end_time.month()).expect("Month not a u8"),
+        u8::try_from(end_time.day()).expect("Day not a u8"),
+        u8::try_from(end_time.hour()).expect("Hour not a u8"),
+        u8::try_from(end_time.minute()).expect("Minute not a u8"),
+        u8::try_from(end_time.second()).expect("Second not a u8"),
+    )
+    .expect("Could not create a not_after time");
+
+    serial[0] += 1;
+
+    let mut builder = Builder::new();
+    let signer_cert_pem = builder
+        .subject_with_nul(SIGNER_SUBJECT)
+        .expect("Could not set subject")
+        .issuer_with_nul(ROOT_SUBJECT)
+        .expect("Could not set issuer")
+        .basic_constraints(false, None)
+        .expect("Could not set basic constraints")
+        .key_usage(KeyUsage::DIGITAL_SIGNATURE | KeyUsage::NON_REPUDIATION)
+        .expect("Could not set key usage")
+        .validity(signer_not_before, signer_not_after)
+        .expect("Could not set time validity range")
+        .serial(&serial[..])
+        .expect("Could not set serial number")
+        .subject_key(&mut signer_subject_key)
+        .issuer_key(&mut signer_issuer_key)
+        .signature_hash(HashType::Sha256)
+        .write_pem_string(&mut csprng)
+        .expect("Could not create PEM string of certificate");
+
+    write(chain_path, root_cert_pem + &signer_cert_pem).expect("Unable to write cert chain");
+}
+
+/// Returns true of the build script should generate the sim files
+fn should_generating_sim_files(data_path: impl AsRef<Path>) -> bool {
+    let data_path = data_path.as_ref();
+    let signal_file = data_path.join("delete_me_to_regenerate_sim_files.empty");
+    cargo_emit::rerun_if_env_changed!("MC_SEED");
+    cargo_emit::rerun_if_changed!(signal_file.to_str().expect("Could not stringify path"));
+
+    // We use a signal file to know if this build script needs to be run again
+    // Using `rerun_if_changed` will ensure this build script runs when the
+    // signal file doesn't exist.
+    // However, the creation/updating of the signal file by this build script is
+    // itself a modification. So we only concern ourselves with the existence of
+    // the signal file.
+    if !signal_file.exists() {
+        write(&signal_file, "").expect("Could not write signal file");
+        return true;
+    }
+
+    // If we got this far we assume that the MC_SEED environment variable was
+    // changed The downside to this approach is that if the MC_SEED is set
+    // initially there will always be one incremental build:
+    // - first build the `signal_file` is created, which tells cargo to re-run this
```

Comment:
I was thinking about this some more over the weekend.
I'm wondering why these files are generated into the repo itself.  I think it's so that the untrusted and the enclaves can use the same files.
But I think it would be better to place these in `$CARGO_TARGET_DIR/sim` like how the enclaves create dedicated directories.

As in general tany of these files could inadvertently be commited to github.

Reply:
I'm not sure why that is. This code has been here for a very very long time and `cargo` went through multiple changes and improvements since it was first written...
It might no longer be the case that there's a good reason for it, or maybe it encodes some learnings that did not get properly documented. Only one way to find out :/

Reply:
I churned more than I should have. I'm pretty sure that it's to share the certs with the main workspace build and the enclaves.

The problem of placing the data files down in the build dir, arises in that there is no common build dir form the enclave's perspective.  The enclaves only know about their version of `CARGO_TARGET_DIR` and `OUT_DIR` and there isn't an easy way for them to correlate to the the main repo's `CARGO_TARGET_DIR` and `OUT_DIR`. It would require either updating the enclave build utility to propagate the main root dir or the logic in this module would have to know about the enclave build directories.  IFF we wanted to invest I would suggest the main root dir is given to the enclave builds, but at this point I don't think we get much benefit updating and can continue on the current path.